### PR TITLE
[ADP-3386] Update sanchonet support

### PR DESCRIPTION
**Pull request by paolino**
- [x] Create structure to run the wallet in many flavors under `run` hierarchy 
- [x] Add script for running a wallet in sanchonet with docker and nix 
- [x] Add an optional E2E test that fully sync the sanchonet using the wallet as monitor 

ADP-3386

### CODE REVIEW

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenTopology.hs
Comment (HeinrichApfelmus) [inline lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenTopology.hs:1]: The pull request touches a huge number of files (600+), but the changes in this file are not, strictly speaking, on topic for this pull request. 🤔  I think that these changes are ok, and happy to merge, but could you split off the 600+ removals into a different pull request for any additional work on top?

**Review by HeinrichApfelmus — APPROVED** [review]
Thank you! 😊 The PR is getting large — 600+ files affected. If you want to do additional work on this — could you either merge this PR or split off the deletions?

**Review by paweljakubas — APPROVED** [review]
wow, not a small at all imprint in codebase!

File: lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenTopology.hs
Comment (paolino) [reply lib/local-cluster/lib/Cardano/Wallet/Launch/Cluster/Node/GenTopology.hs:1]: #4678

**Pull request merged**